### PR TITLE
Feature/415870 access tab changes

### PR DESCRIPTION
**Pull request by PritiPiyushB**
### What one thing this PR does?

Access tab - changes to Resource locator and Contact information elements

### Task details

- [415870 - Access tab - changes to Resource locator and Contact information elements](https://dev.azure.com/defragovuk/DEFRA-NCEA/_workitems/edit/415870/)
- [419062 - Missing 2 fields  Contact information and  NCEA Catalogue Number  from Access Tab page and few more issues](https://dev.azure.com/defragovuk/DEFRA-NCEA/_workitems/edit/419062/)

### Other notes

- Additional details about the task
- Important points that other teams or reviewers should be aware of.

### How do these changes look like?

| name of the screengrab |
Screenshot 2024-08-06 at 13 54 06

### Dev-tested in

1. Local environment

- [Page name/link 1](http://localhost:3000/search/Marine_Scotland_FishDAC_1010?q=marine&jry=qs&pg=1&rpp=20&srt=best_match#access)
- [Page name/link 2](url)

2. Dev environment

- [Page name/link 1](url)
- ...

### CODE REVIEW

File: src/utils/getAccessTabData.ts [hunk 1]
```diff
@@ -5,6 +5,21 @@ import { IAccessItem } from '../interfaces/searchResponse.interface';
...
 
+const rolePrecedence = ['owner', 'pointOfContact', 'custodian', 'distributor', 'originator'];
+
+const combineAndSortContacts = (contactForResource: any[], contact: any[]): any[] => {
+  const contactForResourceArray = contactForResource ?? [];
+  const contactArray = contact ?? [];
+  const combined = [...contactForResourceArray, ...contactArray];
+  combined.sort((a, b) => {
+    const roleIndexA = rolePrecedence.indexOf(a.role.toLowerCase());
+    const roleIndexB = rolePrecedence.indexOf(b.role.toLowerCase());
```
Comment (karthigayayiniram) [inline src/utils/getAccessTabData.ts:16]: rolePrecedence has camelcase but looking for index of role  after transforming them to lowercase. how does it works?

File: src/utils/getAccessTabData.ts [hunk 2]
```diff
@@ -5,6 +5,21 @@ import { IAccessItem } from '../interfaces/searchResponse.interface';
 import { getOrganisationDetails } from './getOrganisationDetails';
 import { addSpaces, capitalizeWords } from './formatAggregationResponse';
 
+const rolePrecedence = ['owner', 'pointOfContact', 'custodian', 'distributor', 'originator'];
+
+const combineAndSortContacts = (contactForResource: any[], contact: any[]): any[] => {
```
Comment (karthigayayiniram) [inline src/utils/getAccessTabData.ts:10]: will this be any? can't we define a interface for this?

File: src/utils/getAccessTabData.ts [hunk 3]
```diff
@@ -55,12 +70,36 @@ const getResourceLocators = (searchItem: Record<string, any>): string => {
...
+  const contactInformationArray: string[] = [];
+
+  const sortedArrayAndCombinedContact = combineAndSortContacts(
+    searchItem?._source?.contact,
+    searchItem?._source?.contactForResource,
+  );
+
+  if (sortedArrayAndCombinedContact.length > 0) {
+    sortedArrayAndCombinedContact.forEach((contact) => {
+      const contactInfo = contact.email
```
Comment (karthigayayiniram) [inline src/utils/getAccessTabData.ts:89]: as contact is of type any. either define the interface or do a safe reading here

File: src/utils/getAccessTabData.ts (hunk 1 above)
Comment (PritiPiyushB) [reply src/utils/getAccessTabData.ts:16]: I have resolved by removing camelcase of pointofcontact.

File: src/utils/getAccessTabData.ts (hunk 2 above)
Comment (PritiPiyushB) [reply src/utils/getAccessTabData.ts]: Added interface for contact

File: src/utils/getAccessTabData.ts (hunk 3 above)
Comment (PritiPiyushB) [reply src/utils/getAccessTabData.ts:89]: Resolved by adding interface

**Pull request merged**